model/event/DacRootAddedEvent.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA
+ *
+ */
+
+namespace oat\taoDacSimple\model\event;
+
+use JsonSerializable;
+use oat\oatbox\event\Event;
+
+/**
+ * Class DacRootAddedEvent This event not triggered on adding child permissions with recursive option
+ * @package oat\taoDacSimple\model\event
+ */
+class DacRootAddedEvent implements Event, JsonSerializable
+{
+
+    /** @var  string */
+    protected $userUri;
+    /** @var string */
+    protected $resourceUri;
+    /** @var string[]  */
+    protected $privilege;
+
+    /**
+     * @param string   $userUri
+     * @param string   $resourceUri
+     * @param string[] $privilege
+     */
+    public function __construct(string $userUri, string $resourceUri, array $privilege)
+    {
+        $this->userUri = $userUri;
+        $this->resourceUri = $resourceUri;
+        $this->privilege = $privilege;
+    }
+
+
+    /**
+     * Return a unique name for this event
+     * @see \oat\oatbox\event\Event::getName()
+     */
+    public function getName()
+    {
+        return get_class($this);
```

Comment:
I don't think this class going to be extended so `__CLASS__` could be ok, and make it `final` just in case :)

Reply:
I'm thinking to create one abstract event and extend 4 events from it. Or extend 2 new events from old. We have 4 events with almost the same content:
`DacRootRemovedEvent` - new. only for 1 resource (not counting recursive)
`DacRootAddedEvent` - new. only for 1 resource (not counting recursive)
`DacRemovedEvent` - old for all changed permissions (recursive permissions)
`DacAddedEvent` - old for all changed permissions (recursive permissions)